gordo_components/builder/azure_utils.py

```diff
@@ -0,0 +1,192 @@
+from typing import Any, Iterator, Tuple, Union
+from datetime import datetime
+from pytz import UTC
+import logging
+
+from mlflow.entities import Metric, Param
+from mlflow.tracking import MlflowClient
+from azureml.core import Workspace
+from azureml.core.authentication import (
+    InteractiveLoginAuthentication,
+    ServicePrincipalAuthentication,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def get_mlflow_client(azureml_config: dict = {}) -> MlflowClient:
+    """
+    Set remote tracking URI for mlflow to AzureML workspace
+
+    Parameters
+    ----------
+    azure_config: dict
+        AzureML Workspace configuration to use for remote MLFlow tracking. Passing only
+        Workspace parameters with a missing our empty `auth` key will result in
+        interactive authentication to AzureML.
+
+        Example::
+
+            `{
+                 "subscription_id":<value>,
+                 "resource_group":<value>,
+                 "workspace_name":<value>
+                 "auth": {
+                     "tenant_id":<value>,
+                     "service_principal_id":<value>,
+                     "service_principal_password":<value>
+                        }
+             }`
+
+    Returns
+    -------
+    client: mlflow.tracking.MlflowClient
+        Client with tracking uri set to AzureML if configured.
+    """
+    logger.info("Creating MLflow tracking client.")
+
+    # Get AzureML tracking_uri if using Azure as backend
+    tracking_uri = None
+    if azureml_config:
```

Comment:
Can this if test ever not trigger? Unless someone explicitly sends in azureml_config = None / False?

I would perhaps expect that it was default "None", and if so, use "local" storage (the default mlflow folder or whatever it is).

Reply:
Empty dicts evaluate to False, so things here behave as suggested.